pkg-py/tests/playwright/chat/llama-index/basic/app.py

```diff
@@ -0,0 +1,46 @@
+from dotenv import load_dotenv
+from llama_index.core.llms import ChatMessage
+from llama_index.llms.openai import OpenAI
+from shiny.express import ui
+
+# Load environment variables from .env file
+_ = load_dotenv()
+
+llm = OpenAI(
+    model="gpt-4o-mini",
+)
+
+ui.page_opts(
+    title="Shiny Chat with LlamaIndex",
+    fillable=True,
+    fillable_mobile=True,
+)
+
+
+chat = ui.Chat(
+    id="chat",
+    messages=[
+        {"role": "system", "content": "You are a pirate with a colorful personality."},
+        {"role": "user", "content": "What is your name, pirate?"},
+        {
+            "role": "assistant",
+            "content": "Arrr, they call me Captain Cog, the chattiest pirate on the seven seas!",
+        },
+    ],
+)
+chat.ui()
+
+
+async def get_response_tokens(conversation: list[ChatMessage]):
+    response_stream = await llm.astream_chat(conversation)
+    async for r in response_stream:
+        yield r.delta
+
+
+@chat.on_user_submit
+async def handle_user_input():
+    conversation = [
+        ChatMessage(role=msg["role"], content=msg["content"]) for msg in chat.messages()
+    ]
```

Comment:
Retaining message history in this way works if the UI & server history is identical, but that's not always true. 

It'd be "more correct" to use llama index's notion of state/context instead https://docs.llamaindex.ai/en/stable/understanding/agent/state/